### ISSUE TITLE
Update MSRV to one that supports newer rusqlite, cbindgen

### ISSUE THREAD

**Issue by djmitche** [GothenburgBitFactory/taskwarrior#2802]
https://github.com/GothenburgBitFactory/taskwarrior/pull/2788
```
error[E0658]: const generics are unstable
```

https://github.com/GothenburgBitFactory/taskwarrior/pull/2789
```
error[E0658]: the `unsafe_op_in_unsafe_fn` lint is unstable
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/os_str_bytes-6.0.1/src/lib.rs:159:1
    |
159 | #![warn(unsafe_op_in_unsafe_fn)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #71668 <https://github.com/rust-lang/rust/issues/71668> for more information

error[E0658]: use of unstable library feature 'slice_strip'
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/os_str_bytes-6.0.1/src/raw_str.rs:606:16
    |
606 |         self.0.strip_prefix(pat).map(Self::from_raw_bytes_unchecked)
    |                ^^^^^^^^^^^^
    |
    = note: see issue #73413 <https://github.com/rust-lang/rust/issues/73413> for more information
```

/cc @dbr

**Comment by djmitche** [comment]
`unsafe_op_in_unsafe_fn` became stable in 1.52.0.

`slice_strip` became stable in 1.51.0.

**Comment by tbabej** [comment]
I believe this was meant to be closed by #2805, @djmitche correct me please if I made the wrong call here (as I had to write the squashed commit message) :)

**Comment by dbr** [comment]
Updating to 1.52 seems perfectly reasonable/conservative :+1: 

A brief look at the cross section of
https://repology.org/project/rust/versions
https://repology.org/project/taskwarrior/versions
doesn't show any obvious problems. Only one that stands out Debian 11 is still on rust/1.48 but I suspect it would only be packaged for 12 anyway
